src/features/events_list/components/FeedSelector/index.tsx

```diff
@@ -0,0 +1,12 @@
+import { useAtom } from '@reatom/react';
+import { featureFlagsAtom, FeatureFlag } from '~core/shared_state';
+import { FeedSelector } from './FeedSelector';
+
+export function FeedSelectorFlagged() {
+  const [featureFlags] = useAtom(featureFlagsAtom);
+  const feedSelectorEnabled =
+    featureFlags[FeatureFlag.FEED_SELECTOR] ||
+    featureFlags[FeatureFlag.EVENTS_LIST__FEED_SELECTOR];
+
+  return feedSelectorEnabled && <FeedSelector />;
+}
```

Comment:
Wouldn't it be better to put this logic into <FeedSelector /> itself? We don't have a feature flag wrapper per-feature elsewhere, and it's expecter from <FeedSelector /> to return `null` if it's not enabled

Reply:
> Wouldn't it be better to put this logic into itself? We don't have a feature flag wrapper per-feature elsewhere

The main reason I did this is because I can't put a useAtom hook inside an if, that would be a violation of the react hook rule. But I need to do it so I don't have to make an extra request to the api when I don't need to

I also can't check this flag on the top level inside the view - since the feature is nested. 
It seems that we invented widgets to integrate several features, but the only widget I see now doesn't import any of the features (now I'm not sure if we need widgets at all).

As a result, it was decided to make the check as a separate component.